sdk/communication/Azure.Communication.Rooms/src/Models/CommunicationIdentifierSerializer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Azure.Communication.Rooms.Models
+{
+    internal class CommunicationIdentifierSerializer
+    {
+        public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel identifier)
+        {
+            string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));
+
+            AssertMaximumOneNestedModel(identifier);
```

Comment:
Ted, let me explain what we are validating here:
A Communication Identifier helps to uniquely identify a room participant.
A communication identifier can either be a Communication User or a Phone number or a Teams user.
These are unique identifiers and there cannot be more than one identifier in a communication Identifier. 
One user can have multiple communication identifiers, but one communication identifier is uniquely 
identified by the type of the identifier.
[These validations are cloned form the communication common shared file].
And safe and better to do on client side to avoid cost of service call failure and fail early.
Let me initiate a chat with someone from communication common to help add more context and 
clarification here.